packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -425,9 +425,12 @@ export default util.createRule({
 
           /* istanbul ignore next */
           default:
-            throw new Error(
-              `Unexpected member property type: ${node.object.type}`,
-            );
+            if (![AST_NODE_TYPES.Identifier, AST_NODE_TYPES.ThisExpression].includes(node.object.type)) {
```

Comment:
A few lines above, there's a comment about `ALLOWED_NON_COMPUTED_PROP_TYPES`. But this change still works despite not adding to that list. I'll defer to @bradzacher as the person who wrote that comment. 🤔

Reply:
So essentially the issue is that #5594 refactored the logic, meaning that `getText` gets called in new ways for a variety of new nodes.

This means that it gets called for more cases than originally intended.
Previously the call to `getText` was always guarded behind `isValidChainTarget` which (I believe) would have enforced the expected `ALLOWED_NON_COMPUTED_PROP_TYPES` constraint.

This is why the code specifically throws an exception here - because based on the prior logic it was not physically possible for any other case to ever get there!

Considering the code now has an unguarded call to `getText`, this switch/case needs to be updated to handle all possible cases!